source/configuration_manual/namespace/index.rst

```diff
@@ -1,84 +1,101 @@
-shared or public. Unfortunately most IMAP clients don't support this extension.
+purpose is to provide Namespace IMAP extension (`RFC
+2342 <http://www.faqs.org/rfcs/rfc2342.html>`_) support, which allows giving
+IMAP clients hints about where to locate mailboxes and whether they're private,
+shared or public.
 
 Dovecot namespaces can be used for several other purposes too:
 
-* Changing the hierarchy separator
+* Changing the :ref:`namespace-hierarchy-separators`
 * Providing backwards compatibility when switching from another IMAP server
-* Provides support for :ref:`public <public_shared_mailboxes>` and :ref:`shared <user_shared_mailboxes>` mailboxes
-* Allows having mails in multiple different locations with possibly different formats
+* Provides support for :ref:`public <public_shared_mailboxes>` and
+  :ref:`shared <user_shared_mailboxes>` mailboxes
+* Allows having mails in multiple different locations with possibly different
+  formats
 
 Configuration
-^^^^^^^^^^^^^^
+=============
 
-There's a default inbox namespace added in `10-mail.conf`. If the configuration
-doesn't explicitly specify a namespace (as was in v2.0 and older) a default
-namespace is created automatically.
+If the Dovecot configuration doesn't explicitly specify a namespace, the
+inbox namespace is created automatically.
 
-The section name in namespaces `(e.g. namespace sectionname { .. }`  is used
-only internally within configuration. It's not required at all, but it allows
-you to update an existing namespace (like how ``15-mailboxes.conf`` does) or
-have userdb override namespace settings for specific users
-``namespace/sectionname/prefix=foo/``.
+Namespace configuration is defined within a dovecot configuration block with
+the format::
 
-All namespace settings with defaults::
+  namespace <section-name> {
+    [... namespace settings ...]
+  }
+
+The optional section name is only used internally within configuration. It
+allows you to update an existing namespace - by repeating the namespace block
+and adding additional configuration settings - or allows userdb to override
+namespace settings for specific users, e.g.::
+
+  namespace/section-name/prefix=foo/
+
+Example configuration for default namespace::
 
   namespace inbox {
-    disabled = no
-    hidden = no
-    ignore_on_failure = no
-    inbox = no
-    list = yes
-    location =
-    order = 0
+    separator = .
     prefix =
-    separator =
-    subscriptions = yes
-    type = private
+    inbox = yes
   }
 
+Namespace Settings
+==================
 
-.. _setting-namespace_name:
+.. _setting-namespace_alias_for:
 
-``namespace name``
-------------------
+``namespace/alias_for``
+-----------------------
 
 - Default: <empty>
-- Values: :ref:`string`
+- Values:  :ref:`string`
+
+If multiple namespaces point to the same location, they should be marked as
+aliases against one primary namespace. This avoids duplicating work for some
+commands (listing the same mailbox multiple times).
```

Comment:
Maybe also some explanation of how this is even possible. Like saying that the alias namespaces often have hidden=yes and list=no so they're not visible unless the clients have specifically configured them, and they're usually used when migrating to a different namespace prefix for existing users.

Reply:
Put this in an info block